### DISCUSSION TITLE
GPS MAP WIDGET

### DISCUSSION THREAD

**Discussion by zachrainnay** [Serial-Studio/Serial-Studio#184] [Q&A] [answered]
Hello Everyone, I have utilized Serial Studio for quite some time to analyze my car's data, I've found great utility in its widgets. Currently, I'm attempting to integrate the GPS widget into my analyses. However, I've encountered an issue where the map area on the GPS visualization appears transparent. I've attempted to upload a GPS data CSV file to rectify this, but unfortunately, the visualization isn't rendering correctly. Could anyone provide guidance on how to resolve this matter and ensure the GPS visualization displays accurately?

Here is a snippet of the GPS data CSV file:
`/*something,-37.6726,144.7547,130.8,0*/`

/*
`something` = Test data
`-37.6726` = Latitude
`144.7547` = Longitude
`130.8` = Altitude
`0` = Test data
*/

pretty sure I have put the data on the right index because it shows on the photo i sent that it is in the right position, i have also tried to switch the index of the data but the same thing happened.

Here is what I meant by it's transparant:
![Screenshot 2024-03-27 014351](https://github.com/Serial-Studio/Serial-Studio/assets/88269879/6c018bdf-39fa-4e40-ab4d-55e43b1298f9)

**Comment by alex-spataru** [comment]
Hi, the problem is with the GPS widget itself, I'll try to solve this issue ASAP during the weekend. I'll keep you notified.

**Comment by alex-spataru** [comment]
Hey there, 

I just updated the codebase to make the map widget functional again. The solution is not perfect, and I still need to register an OpenStreetMap API key for Serial Studio. However, it is a lot better than the current state of the GPS map widget.

![Screenshot_20240401_024004](https://github.com/Serial-Studio/Serial-Studio/assets/4225542/9f9ee339-b6b7-46cc-85ce-b64b1ba42d6a)

You can give it a try in the latest build over at https://github.com/Serial-Studio/Serial-Studio/releases/tag/continuous. Please let me know your feedback on how it's working for you.

By the way, thanks for using Serial Studio for your car data analysis! Do you use any specific OBD2 adapter? Just curious, I recently began diagnosing and fixing car issues with an Autel Maxicom scanner, but it didn't cross my mind to try to use my own software for that 😆

**Reply by egarridriv** [reply]
Hello Alex! Our GPS widget is not working properly with the new version.

**Comment by egarridriv** [comment]
![serialstudiogps](https://github.com/Serial-Studio/Serial-Studio/assets/163044592/74fdd113-1fdd-491c-a65f-229e285941d2)
Our GPS widget is not working properly with the new version.
what are we doing wrong?

**Reply by alex-spataru** [reply]
Hi! 

Please download the **continous** build from this link:

- [https://github.com/Serial-Studio/Serial-Studio/releases/tag/continuous](https://github.com/Serial-Studio/Serial-Studio/releases/tag/continuous)

The current stable build does not contain the changes that I implemented to fix the GPS widget issue.

Thanks,
Alex

**Reply by egarridriv** [reply]
Fantastic! It works very well.
Thank you very much for your great work
![serialstudio gps](https://github.com/Serial-Studio/Serial-Studio/assets/163044592/64edf1a4-5862-4ad7-bcd2-a611962e30e2)

**Reply by zachrainnay** [reply]
Hi! i'd like to ask what's the format of your LAT, LON and ALT data ?

**Reply by alex-spataru** [reply]
Hey @zachrainnay!

I just uploaded an [example project](https://github.com/Serial-Studio/Serial-Studio/tree/master/examples/TinyGPS) that includes both the Arduino code and a Serial Studio project to visualize GPS data on a map. The screenshots in the example are from the [continuous/daily release](https://github.com/Serial-Studio/Serial-Studio/releases/tag/continuous), which will eventually become the next version of Serial Studio.

Right now, I’m in the process of finalizing the new version, and I’ve started writing more documentation and examples. The example projects will be added directly to the source code repository for easier updates, while the wiki will focus on explaining how Serial Studio works, with links pointing to these example projects.

I’m also planning to add many more examples to cover different use cases. Let me know if this example project is helpful or if you have any suggestions! 😊

**Comment by alex-spataru** [comment]
For the API key issues, I think that it might be a good idea to let users define it in the project editor:

<img width="1082" alt="Screenshot 2024-09-16 at 2 00 13 a m" src="https://github.com/user-attachments/assets/d6a87b77-6e4c-4281-af22-cb8568b32776">

I am thinking about adding support for [Thunderforest](https://www.thunderforest.com/maps/), which has some nice maps. Will post an update on this thread when I finish revamping Serial Studio and have a continous release available.

**Comment by alex-spataru** [comment, marked as answer]
Support has just been added for **Thunderforest Maps** and **MapTiler (satellite view)**. To use these features, it is necessary to obtain API keys for each provider and set them in the Project Editor. This feature will be available in the next official release, but it can already be tested in the [continuous release](https://github.com/Serial-Studio/Serial-Studio/releases/tag/continuous).

**Setting API Keys in the Project Editor:**

<img width="1082" alt="Screenshot 2024-09-29 at 1 42 31 p m" src="https://github.com/user-attachments/assets/447b0696-62e1-4676-aaed-95734ba497b3">

**MapTiler Satellite View**  
*(Note: Loading times may vary and can be slow at times.)*

<img width="1185" alt="Screenshot 2024-09-29 at 1 41 15 p m" src="https://github.com/user-attachments/assets/99f0565a-d265-4671-8844-83453f1bc62a">

**Thunderforest Terrain View:**

<img width="1185" alt="Screenshot 2024-09-29 at 1 41 21 p m" src="https://github.com/user-attachments/assets/108d48dc-900f-4860-aa67-14ac3dcfe039">

**Default OpenStreetMap View** *(No API key required)*

<img width="1185" alt="Screenshot 2024-09-29 at 1 41 32 p m" src="https://github.com/user-attachments/assets/b527aebe-1c2f-4e5f-b27c-b1997456d658">

A special thank you to the developers of [SDRangel](https://www.sdrangel.org/) for [creating a class](https://github.com/Serial-Studio/Serial-Studio/blob/master/app/src/Misc/OsmTemplateServer.h) that allows setting API keys for map providers without needing to set up a remote tile server.

**Comment by zachrainnay** [comment]
Hey, I’d like to provide an update on the GPS MAP widget—everything is working perfectly! I know this is a late update, but I just wanted to share the good news. Your explanation was amazing, and your wiki helped a lot. I'd like to close this discussion now.

![GPS MAP WIDGET](https://github.com/user-attachments/assets/a2fba442-4c09-48f2-8f79-a8b3ee2bf938)
